packages/core/src/pubsub/pubsub.ts

```diff
@@ -54,10 +55,11 @@ export class Pubsub extends Observable<PubsubMessage> {
     private readonly topic: string,
     private readonly resubscribeEvery: number,
     private readonly pubsubLogger: ServiceLogger,
-    private readonly logger: DiagnosticsLogger
+    private readonly logger: DiagnosticsLogger,
+    readonly tasks: TaskQueue = new TaskQueue()
   ) {
     super((subscriber) => {
-      const incoming$ = new IncomingChannel(ipfs, topic, resubscribeEvery, pubsubLogger, logger)
+      const incoming$ = new IncomingChannel(ipfs, topic, resubscribeEvery, pubsubLogger, logger, tasks)
```

Comment:
now that we seem to be passing in a shared task queue from higher levels down into each lower level, should we remove the default value for the `tasks` arg on `IncomingChannel` and similar to force callers to pass the queue explicitly?

Reply:
(note: doing that in this PR is totally optional, it's not a priority just wondering about a potential tightening of the API to make it harder to accidentally misuse)